File: src/client/src/types/payment.ts
export enum PaymentMethod {
  Cash = 0,
  CreditCard = 1,
  DebitCard = 2,
  BankTransfer = 3,
  Cheque = 4,
  Insurance = 5,
  Other = 6,
}

export enum PaymentStatus {
  Pending = 0,
  Completed = 1,
  Failed = 2,
  Refunded = 3,
}

export interface PaymentDetail {
  id: string;
  paymentNumber: string;
  invoiceId: string;
  invoiceNumber: string;
  patientId: string;
  patientName: string;
  amount: number;
  paymentMethod: PaymentMethod;
  paymentDate: string;
  referenceNumber?: string | null;
  notes?: string | null;
  receivedBy: string;
  status: PaymentStatus;
  createdAt: string;
  updatedAt?: string | null;
}

export interface PaymentListItem {
  id: string;
  paymentNumber: string;
  patientName: string;
  invoiceNumber: string;
  amount: number;
  paymentMethod: PaymentMethod;
  paymentDate: string;
  status: PaymentStatus;
}

export interface CreatePaymentRequest {
  invoiceId: string;
  amount: number;
  paymentMethod: PaymentMethod;
  referenceNumber?: string | null;
  notes?: string | null;
}

File: src/Server/Infrastructure/Persistence/Entities/Payment.cs
using Server.Core.Common;
using Server.Infrastructure.Persistence.Entities.Enums;

namespace Server.Infrastructure.Persistence.Entities;

public class Payment : BaseEntity
{
    public string PaymentNumber { get; set; } = string.Empty;

    public Guid InvoiceId { get; set; }

    public Guid PatientId { get; set; }

    public decimal Amount { get; set; }

    public PaymentMethod PaymentMethod { get; set; }

    public DateTime PaymentDate { get; set; } = DateTime.UtcNow;

    public string? ReferenceNumber { get; set; }

    public string? Notes { get; set; }

    public Guid ReceivedBy { get; set; }

    public PaymentStatus Status { get; set; } = PaymentStatus.Completed;

    public Invoice? Invoice { get; set; }

    public Patient? Patient { get; set; }
}
